ui/components/app/detected-token/detected-token.js

```diff
@@ -0,0 +1,127 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { useSelector, useDispatch } from 'react-redux';
+
+import {
+  importTokens,
+  ignoreTokens,
+  setNewTokensImported,
+} from '../../../store/actions';
+import { getDetectedTokensInCurrentNetwork } from '../../../selectors';
+
+import DetectedTokenSelectionPopover from './detected-token-selection-popover/detected-token-selection-popover';
+import DetectedTokenIgnoredPopover from './detected-token-ignored-popover/detected-token-ignored-popover';
+
+const DetectedToken = ({ setShowDetectedTokens }) => {
+  const dispatch = useDispatch();
+
+  const detectedTokens = useSelector(getDetectedTokensInCurrentNetwork);
+
+  const [tokensListDetected, setTokensListDetected] = useState(() =>
+    detectedTokens.reduce((tokenObj, token) => {
+      tokenObj[token.address] = { token, selected: true };
+      return tokenObj;
+    }, {}),
+  );
+  const [
+    showDetectedTokenIgnoredPopover,
+    setShowDetectedTokenIgnoredPopover,
+  ] = useState(false);
+
+  const handleClearTokensSelection = async () => {
+    const selectedTokens = Object.values(tokensListDetected)
+      .filter(({ selected }) => {
+        return selected;
+      })
+      .map(({ token }) => {
+        return token;
+      });
+    const unSelectedTokens = Object.values(tokensListDetected)
+      .filter(({ selected }) => {
+        return !selected;
+      })
+      .map(({ token }) => {
+        return token;
+      });
```

Comment:
```suggestion
    // create a lodash chain on this object
    const { selected, deselected } = chain(tokensListDetected)
      // get the values
      .values()
      // create a new object with keys 'selected', 'deselected' and group the tokens
      .groupBy((token) => (token.selected ? 'selected' : 'deselected'))
       // ditch the 'selected' property and get just the tokens'
      .mapValues((group) => group.map(({ token }) => token))n
      // Exit the chain and get the underlying value, an object.
      .value();
```

you could also do a reduce call if you prefer, but lets reduce the amount of times we have to iterate over the tokens. right now there are 4n operations. The above approach is 2n + 2, a reduce statement would be n.